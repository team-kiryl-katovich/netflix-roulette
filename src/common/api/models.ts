export type Result<T> = {
  data?: T;
  error?: Error;
};

export type Offset = {
  offset?: number;
  limit?: number;
};

export type OffsetResult<T> = Result<T> &
  Offset & {
    total?: number;
  };

export type Movie = {
  id: number;
  title: string;
  tagline: string;
  vote_average: number;
  vote_count: number;
  release_date: string;
  poster_path: string;
  overview: string;
  budget: number;
  revenue: number;
  runtime: number;
  genres: string[];
};

export type SortBy = keyof Movie;
export type SortOrder = 'desc' | 'asc';
export type SearchBy = Extract<keyof Movie, 'title' | 'genres'>;

export type MovieFilter = {
  sortBy?: SortBy;
  sortOrder?: SortOrder;
  search?: string;
  searchBy?: SearchBy;
  filter?: string;
};
