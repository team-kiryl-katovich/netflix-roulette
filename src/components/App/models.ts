import { Store } from 'redux';

import { AppState } from '../../store/models';

export type AppProps = {
  store: Store<AppState>;
};
