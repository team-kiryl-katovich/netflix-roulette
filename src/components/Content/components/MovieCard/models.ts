export type MovieCardProps = {
  id: number;
  title: string;
  gengre: string;
  year: number;
  imageUrl: string;
};
