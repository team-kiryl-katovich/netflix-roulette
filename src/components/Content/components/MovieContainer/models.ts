import { WithMoviesProps } from 'src/enhancers/withMovies';

export type MovieContainerComponentProps = WithMoviesProps;
