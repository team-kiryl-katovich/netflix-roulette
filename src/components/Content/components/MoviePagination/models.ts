import { Movie, Offset } from '@common/api';

export type MoviePaginationProps = {
  moviesCount: number;
  offset: Offset;
  pageSize: number;
  loading: boolean;
  children?: React.ReactNode;
  fetch: (offset: Offset) => void;
};

export type MoviePaginationState = {};
