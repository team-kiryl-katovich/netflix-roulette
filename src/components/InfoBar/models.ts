import { WithFilterProps } from '@enhancers/withFilter';
import { WithMoviesProps } from '@enhancers/withMovies';

export type InfoBarComponentProps = WithFilterProps & WithMoviesProps;
