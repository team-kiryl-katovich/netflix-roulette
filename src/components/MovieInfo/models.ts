import { WithMovieProps } from '@enhancers/withMovie';

export type MovieInfoComponentProps = WithMovieProps;
