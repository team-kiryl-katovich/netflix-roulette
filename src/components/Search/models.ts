import { WithMoviesHandlersProps } from '@enhancers/withMovies';
import { WithFilterProps } from '@enhancers/withFilter';

export type SearchComponentProps = WithFilterProps & WithMoviesHandlersProps;
