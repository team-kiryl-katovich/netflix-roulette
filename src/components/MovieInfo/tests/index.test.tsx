import * as React from 'react';
import * as enzyme from 'enzyme';

import { MovieInfoComponentProps } from '../models';
import { MovieInfoComponent } from '..';

jest.mock('react-router-dom', () => ({
  useParams: jest.fn(() => ({
    id: '1',
  })),
}));

describe('[COMPONENTS] [MOVIE_INFO]', () => {
  test('should render component', () => {
    const fakeProps: MovieInfoComponentProps = {
      movieData: {
        error: null,
        loading: false,
        movie: {
          id: 1,
          genres: ['fakeGenre'],
          budget: 1,
          overview: 'fakeOverview',
          poster_path: 'fakePosterPath',
          release_date: '2019-01-01',
          revenue: 1,
          runtime: 1,
          tagline: 'fakeTagline',
          title: 'fakeTitle',
          vote_average: 1,
          vote_count: 1,
        },
        notFound: false,
      },
      selectMovie: jest.fn(),
    };

    const component = enzyme.shallow(<MovieInfoComponent {...fakeProps} />);
    expect(component).toMatchSnapshot();
  });
});
