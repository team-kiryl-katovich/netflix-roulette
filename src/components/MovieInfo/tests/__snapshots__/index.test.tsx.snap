// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS] [MOVIE_INFO] should render component 1`] = `
<div
  className="movie-info"
>
  <Component
    className="movie-info__search-link"
    to="/"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip="horizontal"
      icon="search"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Component>
  <div
    className="movie-info__image"
  >
    <Component
      ratio={140}
      src="fakePosterPath"
    />
  </div>
  <div
    className="movie-info__content"
  >
    <div
      className="movie-info-header"
    >
      <Component
        className="movie-info-header__title"
        text="fakeTitle"
        uppercase={true}
      />
      <Component
        className="movie-info-header__rating"
        text={1}
      />
    </div>
    <Component
      className="movie-info-subtitle"
      text="fakeTagline"
    />
    <div
      className="movie-info-time"
    >
      <div
        className="movie-info-time__year"
      >
        <Component
          className="movie-info-time-label"
          text={2019}
          uppercase={true}
        />
        <Component
          className="movie-info-time-sublabel"
          text="year"
        />
      </div>
      <div
        className="movie-info-time__duration"
      >
        <Component
          className="movie-info-time-label"
          text={1}
        />
        <Component
          className="movie-info-time-sublabel"
          text="min"
        />
      </div>
    </div>
    <Component
      className="movie-info-description"
      text="fakeOverview"
    />
  </div>
</div>
`;
